persistence/src/main/java/org/dswarm/persistence/monitoring/MonitoringLogger.java

```diff
@@ -57,59 +54,114 @@
+	private final MarkedTimer executionsTimer;
+	private final MarkedTimer ingestTimer;
+	private final Logger logger;
+	private final ObjectMapper mapper;
 	private final MetricFilter noSpecialTimer;
-	private static final CharMatcher MATCHER =
-			CharMatcher.ASCII
-					.and(CharMatcher.JAVA_LETTER_OR_DIGIT)
-					.negate()
-					.or(CharMatcher.WHITESPACE);
+	private final long rateFactor;
+	private final TimeUnit rateUnit;
+	private final String rateUnitName;
+	private final MetricRegistry registry;
+	private final List<MarkedTimer> specialTimers;
 
 	@Inject
 	private MonitoringLogger(
 			@Named("Monitoring") final ObjectMapper mapper,
 			@Named("Monitoring") final MetricRegistry registry,
-			@Named("Monitoring") final Logger logger) {
+			@Named("Monitoring") final Logger logger,
+			@Named("dswarm.monitoring.rate-unit") final String rateUnitFromConfig,
+			@Named("dswarm.monitoring.duration-unit") final String durationUnitFromConfig,
+			@Named("dswarm.monitoring.continuous-interval") final long continuousIntervalInMillisFromConfig) {
 		this.mapper = mapper;
 		this.registry = registry;
 		this.logger = logger;
 
-		// TODO: config value
-		final TimeUnit rateUnit = TimeUnit.SECONDS;
-		final TimeUnit durationUnit = TimeUnit.MILLISECONDS;
+		rateUnit = TimeUnit.valueOf(rateUnitFromConfig.toUpperCase());
+		durationUnit = TimeUnit.valueOf(durationUnitFromConfig.toUpperCase());
+		continuousInterval = continuousIntervalInMillisFromConfig;
+
+		rateUnitName = rateUnitFromConfig.substring(0, rateUnitFromConfig.length() - 1);
+		durationUnitName = durationUnitFromConfig;
 
-		this.rateUnit = calculateRateUnit(rateUnit);
-		this.durationUnit = durationUnit.toString().toLowerCase(Locale.US);
 		rateFactor = rateUnit.toSeconds(1);
 		durationFactor = 1.0 / durationUnit.toNanos(1);
 
-		executionsTimerName = name(Task.class, "executions");
-		executionsTimer = registry.timer(executionsTimerName);
+		final String executionsTimerName = name(Task.class, "executions");
+		executionsTimer = new MarkedTimer(
+				executionsTimerName,
+				EXECUTION_MARKER,
+				registry.timer(executionsTimerName));
 
-		ingestTimerName = name(DataModel.class, "ingest");
-		ingestTimer = registry.timer(ingestTimerName);
+		final String ingestTimerName = name(DataModel.class, "ingest");
+		ingestTimer = new MarkedTimer(
+				ingestTimerName,
+				INGEST_MARKER,
+				registry.timer(ingestTimerName));
+
+		specialTimers = ImmutableList.of(executionsTimer, ingestTimer);
 
 		noSpecialTimer = (name, metric) ->
-				!(executionsTimerName.equals(name) || ingestTimerName.equals(name));
+				!specialTimers.stream().anyMatch(mt -> mt.matches(name));
 	}
 
 	public void report() {
+		report(markedTimer -> true);
+	}
+
+	public MonitoringHelper startExecution(final Task task) {
+		final EntityIdentification identification = EntityIdentification.of(task);
+		final MDCCloseable mdc = identification.putMDC();
+
+		task.getJob().getMappings().forEach(this::monitorEntity);
```

Comment:
The overall time is tracked by the MonitoringHelper, cumulative is all entities, records, and literals combined. 


> On Mar 27, 2015, at 10:20 AM, Bo Ferri <notifications@github.com> wrote:
> 
> In persistence/src/main/java/org/dswarm/persistence/monitoring/MonitoringLogger.java:
> 
> >  	}
> >  
> >  	public void report() {
> > +		report(markedTimer -> true);
> > +	}
> > +
> > +	public MonitoringHelper startExecution(final Task task) {
> > +		final EntityIdentification identification = EntityIdentification.of(task);
> > +		final MDCCloseable mdc = identification.putMDC();
> > +
> > +		task.getJob().getMappings().forEach(this::monitorEntity);
> is time of all mapping executions available via cumulative?
> 
> —
> Reply to this email directly or view it on GitHub.
>